app/src/main/java/com/eis/audioUtility/AudioUtilityManager.java

```diff
@@ -0,0 +1,199 @@
+ * and to manage Vibration
+ *
+ * @author Francesco Bau'
+ */
+
+import android.content.Context;
+import android.media.AudioManager;
+import android.os.Build;
+import android.os.VibrationEffect;
+import android.os.Vibrator;
+
+import static android.content.Context.AUDIO_SERVICE;
+
+
+public class AudioUtilityManager {
+
+    public static final int MAX_PERCENTAGE = 100;
+    public static final int MIN_PERCENTAGE = 0;
+
+    private static final int VIBRATION_TIME = 500;
+    private static final int VIBRATION_DELAY = 500;
+    private static final int VIBRATION_REPEAT_CODE = 1;
+
+    public enum AUMStream {
+        ALARM,
+        RING,
+        MUSIC
+    }
+
+    /**
+     * @param context The current Context.
+     * @return The current AudioManager instance.
+     */
+    protected static AudioManager getAudioManager(Context context) {
+        return (AudioManager) context.getSystemService(AUDIO_SERVICE);
+    }
+
+    /**
+     * @param context The current Context.
+     * @param stream  The chosen stream (should be ALARM, RING or MUSIC).
+     * @return The current stream Volume (in percentage).
+     */
+    public static int getVolume(Context context, AUMStream stream) {
+        int currentVolume = getAudioManager(context).getStreamVolume(getStream(stream));
+
+        int maxVolume = getMaxVolume(context, stream);
+        return Math.round(MAX_PERCENTAGE * currentVolume / maxVolume);
+    }
+
+    /**
+     * @param context The current Context.
+     * @param stream  The chosen stream (should be ALARM, RING or MUSIC).
+     * @return The maximum stream volume (real value).
+     */
+    private static int getMaxVolume(Context context, AUMStream stream) {
+        return getAudioManager(context).getStreamMaxVolume(getStream(stream));
+    }
+
+    /**
+     * Sets up the stream Volume, given a certain percentage.
+     *
+     * @param context    The current Context.
+     * @param stream     The chosen stream (should be ALARM, RING or MUSIC).
+     * @param percentage Target volume (expressed in %).
+     * @throws IllegalArgumentException if percentage is not between 0 and 100.
+     */
+    public static void setVolume(Context context, AUMStream stream, int percentage) throws IllegalArgumentException {
+        if (percentage < MIN_PERCENTAGE)
+            throw new IllegalArgumentException(" Your value is too low. Please insert a value between 0 and 100.");
+        if (percentage > MAX_PERCENTAGE)
+            throw new IllegalArgumentException(" Your value is too high. Please insert a value between 0 and 100.");
+        int maxVolume = getMaxVolume(context, stream);
+
+        // Calculate the real value of the new volume
+        int newVolume = maxVolume * percentage;
+        newVolume = Math.round(newVolume / MAX_PERCENTAGE);
+
+        // Sets the Volume.
+        getAudioManager(context).setStreamVolume(
+                getStream(stream),
+                newVolume,
+                AudioManager.FLAG_SHOW_UI
```

Comment:
What if I didn't want to have the user see that I'm modifying the audio volume? You could add a parameter like `notifyUser` that enables or disables this show on UI

Reply:
That's a good idea!
Is that supposed to be a boolean, then?

Reply:
Idk, see AudioManager documentation, it's probably a boolean

Reply:
Not really, the flag is not a boolean... it's an int value.
Sorry for my bad English, what I meant to ask you is: should I add a new boolean parameter in the method? So, if it's true, the flag will be shown, and it would be hidden otherwise.

But we get into another issue by doing this: there are gonna be 4 parameters in this method....
I was thinking about an overload of the method, but the only one I can think about is to set the volume with the default Stream... And, in our app, it should be Alarm, right?